src/openrct2-ui/windows/TrackDesignPlace.cpp

```diff
@@ -598,28 +599,27 @@ static void window_track_place_draw_mini_preview_track(
         }
         if (!(rotation & 4))
         {
-            origin.x += CoordsDirectionDelta[rotation].x;
-            origin.y += CoordsDirectionDelta[rotation].y;
+            origin.x = rotatedAndOfffsetTrack.x + CoordsDirectionDelta[rotation].x;
+            origin.y = rotatedAndOfffsetTrack.x + CoordsDirectionDelta[rotation].y;
```

Comment:
When you fix the type this can be just a simple `+`. Also note you've done .x for the .y coordinate.